builder/src/extensionConfig.ts

```diff
@@ -78,9 +78,11 @@ function generateConfig({
     ...(coreData.resolutions ?? {})
   };
 
+  // Alow extensions to match a wider range than the core dependency
+  // To ensure forward compatibility.
   Object.keys(coreDeps).forEach(element => {
     shared[element] = {
-      requiredVersion: coreDeps[element],
+      requiredVersion: coreDeps[element].replace('~', '^'),
```

Comment:
I'm not sure exactly what this is affecting - does it make sense to not restrict to ~ updates in core now?